beets/util/__init__.py

```diff
@@ -1094,6 +1095,30 @@ def __get__(self, instance, owner):
         return self.cache[owner]
 
 
+T = TypeVar("T")
+
+
+class LazySharedInstance(Generic[T]):
+    """A descriptor that provides access to a lazily-created shared instance of
+    the containing class, while calling the class constructor to construct a
+    new object works as usual.
```

Comment:
Can we add a usage example here?

Is there any benefit in using this pattern instead of the more traditional singleton pattern?

Reply:
If I remember correctly, this was the result of refactoring the existing code (with the `ArtResizer.shared` accessor) in a way that I managed to write typings for and seemed a bit simpler than the previous implementation.

Whether we should have the `ArtResizer.shared` and `ArtResizer()` to get shared and owned instances, or just a singleton is a separate discussion that I'd prefer not to open up in this PR, because it would require changing all uses of the art resizer. Currently, this PR leaves the interface unchanged.

I'll add an example.

Reply:
All right thanks!

I was just wondering why this changed. It is the only none typing part in this PR  in comparison to the previous implementation.

If it works tho, I'm happy with it :+1:

Reply:
There was something similar before (the `Shareable` metaclass), but I moved it from `artresizer.py` to `__init__.py` and rewrote it as "just" a descriptor instead of a full-blown metaclass.